cmd/bbolt/command_surgery_meta.go

```diff
@@ -0,0 +1,317 @@
+		}
+	}
+
+	return nil
+}
+
+func newSurgeryMetaUpdateCommand() *cobra.Command {
+	var o surgeryMetaUpdateOptions
+	metaUpdateCmd := &cobra.Command{
+		Use:   "update <bbolt-file> [options]",
+		Short: "Update fields in meta pages",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if len(args) == 0 {
+				return errors.New("db file path not provided")
+			}
+			if len(args) > 1 {
+				return errors.New("too many arguments")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if err := o.Validate(); err != nil {
+				return err
+			}
+			return surgeryMetaUpdateFunc(args[0], o)
+		},
+	}
+	o.AddFlags(metaUpdateCmd.Flags())
+	return metaUpdateCmd
+}
+
+func surgeryMetaUpdateFunc(srcDBPath string, cfg surgeryMetaUpdateOptions) error {
+	if _, err := checkSourceDBPath(srcDBPath); err != nil {
+		return err
+	}
+
+	if err := common.CopyFile(srcDBPath, cfg.outputDBFilePath); err != nil {
+		return fmt.Errorf("[meta update] copy file failed: %w", err)
+	}
+
+	pageSize := cfg.pageSize
+	if pageSize == 0 {
+		pageSize = uint32(common.DefaultPageSize)
+	}
+
+	// update meta page 0
+	m0, buf0, err := ReadMetaPageAt(cfg.outputDBFilePath, 0, pageSize)
+	if err != nil {
+		return fmt.Errorf("read the first meta page failed: %w", err)
+	}
+	m0Changed := updateMetaField(m0, cfg.fields)
+	if m0Changed {
+		if err := writeMetaPageAt(cfg.outputDBFilePath, buf0, 0, pageSize); err != nil {
+			return fmt.Errorf("[meta update] write the first meta page failed: %w", err)
+		}
+	}
+
+	if pageSize != m0.PageSize() {
+		fmt.Fprintf(os.Stdout, "WARNING: The provided page size (%d) isn't consistent with the page size (%d) in the first valid meta page\n", pageSize, m0.PageSize())
+	}
+
+	// update meta page 1
+	m1, buf1, err := ReadMetaPageAt(cfg.outputDBFilePath, 1, pageSize)
+	if err != nil {
+		return fmt.Errorf("read the second meta page failed: %w", err)
+	}
+
+	m1Changed := updateMetaField(m1, cfg.fields)
+	if m1Changed {
+		if err := writeMetaPageAt(cfg.outputDBFilePath, buf1, 1, pageSize); err != nil {
+			return fmt.Errorf("[meta update] write the second meta page failed: %w", err)
+		}
+	}
+
+	// Display results
+	if !m0Changed && !m1Changed {
+		fmt.Fprintln(os.Stdout, "Nothing changed!")
+	}
+
+	if m0Changed {
+		fmt.Fprintln(os.Stdout, "The first meta page has been updated!")
+	}
+	if m1Changed {
+		fmt.Fprintln(os.Stdout, "The second meta page has been updated!")
+	}
+
+	return nil
+}
+
+func updateMetaField(m *common.Meta, fields []string) bool {
+	changed := false
+	for _, field := range fields {
+		kv := strings.Split(field, ":")
+		val, _ := strconv.ParseUint(kv[1], 10, 64)
+
+		switch kv[0] {
+		case "pageSize":
+			m.SetPageSize(uint32(val))
+		case "root":
+			m.SetRootBucket(common.NewInBucket(common.Pgid(val), 0))
+		case "freelist":
+			m.SetFreelist(common.Pgid(val))
+		case "pgid":
+			m.SetPgid(common.Pgid(val))
+		case "txid":
+			m.SetTxid(common.Txid(val))
```

Comment:
Usually by Txid the bbolt knows which meta-page is the newer (as it has higher Txid). 

Overriding Txid on both pages might bring more problems than value... 

(It's not clear to me that meta operations should default to editing both pages at once... 
I would assume they should explicitly require pointing which page to edit... and user can call them 2 times if the intent is to edit both pages... 
Batch surgeries... cannot end good...
)

Reply:
Updated:
- Removed the Txid from the list (do not allow uses to change txid) although it  (updating txid) will not cause any problem;
- Users must specify a meta page id to operate on. They can only update one meta page at a time.

thx